### PR TITLE
Use crossed() in sample strategy

### LINKED ISSUE #1364: Default strategy has to be changed as it is misleading for new users.
as discussed in #1362 the default strategy has to be changed to a more reasonable one. 
some basic things are not respected in default strategy which can be misleading and discouraging for new users.
Issues I can think of: 
- It doesn't check the volume so price can deviate radically from TA.
- Buy/Sell moments are propagated to many frames.

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
The Sample strategy  and documentation should try to guide users in the right direction when developing strategies, therefore it's important to have good samples.

Using crossed() as signal avoids generating "sticking" buy-signals.
e.g. you want to buy whe ADX crosses 30 from the bottom - not whenever it's > 30.

This is especially problematic with limited max_trades - since it could buy when ADX > 90 - and sell immediately afterwards.

closes #1364

## Quick changelog

- Improve Samplestrategy
- improve and comment strategy-customization documentation

### CODE REVIEW

File: docs/strategy-customization.md [hunk 1]
```diff
@@ -138,15 +138,19 @@ def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
...
+            (dataframe['tema'] > dataframe['tema'].shift(1)) &  # Guard
+            (dataframe['volume'] > 0)  # Make sure Volume is not 0
         ),
         'buy'] = 1
 
     return dataframe
 ```
 
+!!! Note
+    Buying requires sellers to buy from - therefore volume needs to be > 0 (`dataframe['volume'] > 0`) to make sure backtesting does not buy/sell in no-activity periods.
```
Comment (hroff-1902) [inline docs/strategy-customization.md]: "backtesting does not buy"? or it's strategy which buys?

File: docs/strategy-customization.md [hunk 2]
```diff
@@ -138,15 +138,19 @@ def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
     """
     dataframe.loc[
         (
-            (dataframe['adx'] > 30) &
-            (dataframe['tema'] <= dataframe['bb_middleband']) &
-            (dataframe['tema'] > dataframe['tema'].shift(1))
+            (qtpylib.crossed_above(dataframe['adx'], 30)) &  # Signal: ADX crosses baove 30
```
Comment (hroff-1902) [inline docs/strategy-customization.md]: "above 30"

File: docs/strategy-customization.md [hunk 3]
```diff
@@ -168,9 +172,10 @@ def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame
     """
     dataframe.loc[
         (
-            (dataframe['adx'] > 70) &
-            (dataframe['tema'] > dataframe['bb_middleband']) &
-            (dataframe['tema'] < dataframe['tema'].shift(1))
+            (qtpylib.crossed_above(dataframe['adx'], 70)) &  # Signal: ADX crosses above 30
```
Comment (hroff-1902) [inline docs/strategy-customization.md]: "above 70"

File: user_data/strategies/sample_strategy.py [hunk 4]
```diff
@@ -295,9 +295,9 @@ def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame
         """
         dataframe.loc[
             (
-                (dataframe['adx'] > 70) &
+                (qtpylib.crossed_above(dataframe['adx'], 70)) &  # Signal: - ADX crosses above 70
                 (dataframe['tema'] > dataframe['bb_middleband']) &
-                (dataframe['tema'] < dataframe['tema'].shift(1)) &
+                (dataframe['tema'] < dataframe['tema'].shift(1)) &  # Guard: tema is raising
```
Comment (hroff-1902) [inline user_data/strategies/sample_strategy.py]: it's not raising here, it goes down

File: user_data/strategies/sample_strategy.py [hunk 5]
```diff
@@ -295,9 +295,9 @@ def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame
         """
         dataframe.loc[
             (
-                (dataframe['adx'] > 70) &
+                (qtpylib.crossed_above(dataframe['adx'], 70)) &  # Signal: - ADX crosses above 70
```
Comment (hroff-1902) [inline user_data/strategies/sample_strategy.py]: remove '-' character

File: docs/strategy-customization.md [hunk 6]
```diff
@@ -168,9 +172,10 @@ def populate_sell_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame
...
+            (dataframe['tema'] > dataframe['bb_middleband']) &  # Guard
+            (dataframe['tema'] < dataframe['tema'].shift(1)) &  #Guard
```
Comment (hroff-1902) [inline docs/strategy-customization.md]: space after `#`

**Comment by xmatthias** [comment]
> A common note: are you sure crossed_above() is applicable here for ADX

well if the trend wasn't below 30, then you're late / the bot bought it already.

To be honest, i'm not sure if ADX is suited as signal at all... 
It probably works better with RSI (maybe we should switch this sample strategy to RSI ...?) - but it's not "wronger" for ADX than it was before - where no "signal" was created, but a swamp of signals (which buys when ADX > 30, and sells when ADX <70 - which is also true at 31) ...

it generates a few 1000 signals in 5m with ~20 pairs for all of 2018 ...

File: docs/strategy-customization.md [hunk 7]
```diff
@@ -149,7 +149,7 @@ def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
 ```
 
 !!! Note
-    Buying requires sellers to buy from - therefore volume needs to be > 0 (`dataframe['volume'] > 0`) to make sure backtesting does not buy/sell in no-activity periods.
+    Buying requires sellers to buy from - therefore volume needs to be > 0 (`dataframe['volume'] > 0`) to make sure that the does not buy/sell in no-activity periods.
```
Comment (hroff-1902) [inline docs/strategy-customization.md]: "that the bot"

**Comment by xmatthias** [comment]
I've switched this now to RSI since with this configuration, the bot generates profits (so it could be profitable out of the box, assuming correct pairs, correct markets, ...).

```
2019-10-15 19:36:28,154 - freqtrade.optimize.backtesting - INFO - Backtesting with data from 2018-01-01T00:00:00+00:00 up to 2018-12-31T23:55:00+00:00 (364 days)..
2019-10-15 19:36:28,154 - freqtrade.optimize.backtesting - INFO - Running backtesting for Strategy SampleStrategy
Result for strategy SampleStrategy
========================================================= BACKTESTING REPORT ========================================================
| pair      |   buy count |   avg profit % |   cum profit % |   tot profit ETH |   tot profit % | avg duration   |   profit |   loss |
|:----------|------------:|---------------:|---------------:|-----------------:|---------------:|:---------------|---------:|-------:|
| ADA/ETH   |          93 |           0.16 |          14.51 |       0.01452402 |           7.25 | 8:40:00        |       73 |     20 |
| AST/ETH   |         177 |           0.04 |           7.73 |       0.00773533 |           3.86 | 5:56:00        |      125 |     52 |
| BTG/ETH   |         114 |           0.20 |          22.98 |       0.02300359 |          11.49 | 7:48:00        |       87 |     27 |
| DASH/ETH  |          92 |           0.15 |          13.39 |       0.01340455 |           6.70 | 9:39:00        |       64 |     28 |
| EOS/ETH   |         117 |          -0.16 |         -18.67 |      -0.01869357 |          -9.34 | 11:03:00       |       77 |     40 |
| ETC/ETH   |          64 |           0.27 |          17.24 |       0.01725497 |           8.62 | 8:02:00        |       53 |     11 |
| HOT/ETH   |          25 |           0.31 |           7.87 |       0.00788264 |           3.94 | 10:36:00       |       23 |      2 |
| IOTA/ETH  |          59 |           0.06 |           3.69 |       0.00369630 |           1.85 | 7:19:00        |       48 |     11 |
| IOTX/ETH  |          56 |           0.14 |           7.74 |       0.00774610 |           3.87 | 6:59:00        |       42 |     14 |
| LSK/ETH   |         110 |          -0.24 |         -26.30 |      -0.02632948 |         -13.15 | 8:17:00        |       72 |     38 |
| LTC/ETH   |          43 |           0.29 |          12.62 |       0.01263092 |           6.31 | 12:25:00       |       33 |     10 |
| NEO/ETH   |          49 |           0.13 |           6.52 |       0.00652615 |           3.26 | 8:09:00        |       40 |      9 |
| NANO/ETH  |          60 |          -0.32 |         -19.12 |      -0.01914008 |          -9.56 | 5:56:00        |       48 |     12 |
| NCASH/ETH |          79 |           0.08 |           6.33 |       0.00633206 |           3.16 | 8:20:00        |       60 |     19 |
| OMG/ETH   |          61 |          -0.08 |          -5.11 |      -0.00511355 |          -2.55 | 9:38:00        |       47 |     14 |
| TRX/ETH   |          96 |           0.47 |          44.89 |       0.04493928 |          22.45 | 7:32:00        |       80 |     16 |
| VET/ETH   |           8 |           0.31 |           2.48 |       0.00248561 |           1.24 | 5:01:00        |        6 |      2 |
| XLM/ETH   |          50 |          -0.20 |          -9.95 |      -0.00995646 |          -4.97 | 9:59:00        |       38 |     12 |
| XMR/ETH   |          65 |          -0.35 |         -22.94 |      -0.02296497 |         -11.47 | 9:08:00        |       47 |     18 |
| XRP/ETH   |          56 |          -0.87 |         -48.46 |      -0.04851017 |         -24.23 | 9:51:00        |       39 |     17 |
| XVG/ETH   |          59 |          -0.12 |          -7.02 |      -0.00702939 |          -3.51 | 8:31:00        |       45 |     14 |
| ZEC/ETH   |          45 |           0.05 |           2.18 |       0.00217868 |           1.09 | 10:13:00       |       32 |     13 |
| TOTAL     |        1578 |           0.01 |          12.59 |       0.01260253 |           6.29 | 8:27:00        |     1179 |    399 |
========================================================= SELL REASON STATS =========================================================
| Sell Reason   |   Count |
|:--------------|--------:|
| roi           |    1104 |
| sell_signal   |     399 |
| stop_loss     |      73 |
| force_sell    |       2 |
====================================================== LEFT OPEN TRADES REPORT ======================================================
| pair     |   buy count |   avg profit % |   cum profit % |   tot profit ETH |   tot profit % | avg duration    |   profit |   loss |
|:---------|------------:|---------------:|---------------:|-----------------:|---------------:|:----------------|---------:|-------:|
| EOS/ETH  |           1 |          -3.38 |          -3.38 |      -0.00338589 |          -1.69 | 3 days, 8:10:00 |        0 |      1 |
| IOTX/ETH |           1 |          -2.78 |          -2.78 |      -0.00277817 |          -1.39 | 14:00:00        |        0 |      1 |
| TOTAL    |           2 |          -3.08 |          -6.16 |      -0.00616406 |          -3.08 | 1 day, 23:05:00 |        0 |      2 |
```

**Comment by hroff-1902** [comment]
Super 👍 ,
just correct pls the comments -- you left ADX in the comments.

the ADX indicator can also be kept in the comments in populate_indicators, just above Awesome Osc.

File: user_data/strategies/sample_strategy.py [hunk 8]
```diff
@@ -103,8 +102,10 @@ def populate_indicators(self, dataframe: DataFrame, metadata: dict) -> DataFrame
         # ------------------------------------
 
         # ADX
-        dataframe['adx'] = ta.ADX(dataframe)
+        # dataframe['adx'] = ta.ADX(dataframe)
```
Comment (xmatthias) [inline user_data/strategies/sample_strategy.py]: It is ... ?

Comment (hroff-1902) [reply user_data/strategies/sample_strategy.py]: Almost :)

Pls see my variant, sorry that I changed your branch -- that was more effective than explain it in words...

Comment (xmatthias) [reply user_data/strategies/sample_strategy.py]: that's fine :+1: 

ADX is now commented twice (that's what i didn't get) - but that's fine :)

Comment (hroff-1902) [reply user_data/strategies/sample_strategy.py]: oops, and you did not fix it... (double comment)

**Pull request merged**